cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/AbstractDataprocProvisioner.java

```diff
@@ -277,20 +275,26 @@ protected final Map<String, String> createContextProperties(ProvisionerContext c
   /**
    * Returns a set of system labels that should be applied to all Dataproc entities.
    */
-  protected final Map<String, String> getSystemLabels() {
+  protected final Map<String, String> getCommonDataprocLabels(ProvisionerContext provisionerContext) {
     Map<String, String> labels = new HashMap<>();
+    // Add labels from provisioner properties.
+    // labels are expected to be in format:
+    // name1=val1,name2=val2
+    String provisionerLabelsStr = provisionerContext.getProperties().get(LABELS_PROPERTY);
+    if (provisionerLabelsStr != null) {
+      labels.putAll(DataprocUtils.parseLabels(provisionerLabelsStr));
```

Comment:
I believe you need to use parseKeyValueConfig here as `provisionerContext` uses different separators

Reply:
Done.